### PR TITLE
User Login and Registration

### PR DESCRIPTION
**Pull request by davenmathews**
This pull request implements user log in and registration using flask.
It contains methods only for signing up and logging in only.

It finishes the following Pivotal Tracker stories:
#152975928,  #152975928

### CODE REVIEW

File: events.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (ebrahim-j) [inline events.json]: Why this approach compared to just plain data structures? Same for `users.json`

File: events.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (ebrahim-j) [inline events.txt]: What is the purpose of this file + `users.txt
`

File: app.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        u.close()
+    with open(DATA_EVENT_FILE_PATH, mode='w') as e:
+        json.dump([], e)
+        e.close()
+
+
+@app.route('/user_home')
+def user_home():
+    events = get_events_from_file()
+    print(session)
```
Comment (ebrahim-j) [inline app.py]: Remove all `print` statements from code

File: app.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
+from flask import Flask, render_template, redirect, url_for, request, session, flash,jsonify
+from data import User, Event, DATA_USER_FILE_PATH, DATA_EVENT_FILE_PATH
+import json
+
+app = Flask(__name__)
+app.secret_key = "secret_key"
+app.config['SEND_FILE_MAX_AGE_DEFAULT'] = 0
+
+
+# dave = User('Dave', 'Mathews', 'davenmathews@gmail.com', 'password')
```
Comment (ebrahim-j) [inline app.py]: Clean up code. Remove all unnecessary/commented out code

File: data.py [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+                if x['event_id'] == self.get_event_id():
+                    k.remove(event)
+                    break
+            with open(DATA_EVENT_FILE_PATH, mode='w') as p2:
+                json.dump(k, p2)
+                p2.close()
+            p.close()
+
+
+class User(object):
```
Comment (ebrahim-j) [inline data.py]: Separate classes into two different files

**Comment by ebrahim-j** [comment]
I am curious as to why the files `app.py` and `data.py` seem to have been added in **mega** commits

**Comment by ebrahim-j** [comment]
Some of your commit messages don't match the actual stuff that has been committed..
for example 

> Add functionality for user to add and modify events is only changing variable/function names

> Fix bug with data.py is only commenting out some _test_ code

File: events.json (hunk 1 above)
Comment (davenmathews) [reply events.json]: Every time I ran app.py without storing the files, I would have to create new users and events...which meant signing up a new every time the browser restarted

File: events.txt (hunk 2 above)
Comment (davenmathews) [reply events.txt]: I used .txt at first but later changed to .json for readability in vsCode

**Pull request merged**